MdeModulePkg/Library/SmmCorePerformanceLib/StandaloneMmCorePerformanceLib.c

```diff
@@ -0,0 +1,148 @@
+/** @file
+  This library is mainly used by the Standalone MM Core to start performance logging to ensure that
+  Standalone MM Performance and PerformanceEx Protocol are installed as early as possible in the MM phase.
+
+  Caution: This module requires additional review when modified.
+    - This driver will have external input - performance data and communicate buffer in MM mode.
+    - This external input must be validated carefully to avoid security issue like
+      buffer overflow, integer overflow.
+
+  SmmPerformanceHandlerEx(), SmmPerformanceHandler() will receive untrusted input and do basic validation.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include "SmmCorePerformanceLibInternal.h"
+
+#include <Library/StandaloneMmMemLib.h>
+
+extern BOOLEAN  mPerformanceMeasurementEnabled;
+
+/**
+  A library internal MM-instance specific implementation to check if a buffer outside MM is valid.
+
+  This function is provided so Standalone MM and Traditional MM may use a different implementation
+  of data buffer check logic.
+
+  @param[in] Buffer  The buffer start address to be checked.
+  @param[in] Length  The buffer length to be checked.
+
+  @retval TRUE  This buffer is valid per processor architecture.
+  @retval FALSE This buffer is not valid per processor architecture.
+**/
+BOOLEAN
+IsBufferOutsideMmValidInternal (
+  IN EFI_PHYSICAL_ADDRESS  Buffer,
+  IN UINT64                Length
+  )
+{
+  return MmIsBufferOutsideMmValid (Buffer, Length);
+}
+
+/**
+  A library internal MM-instance specific implementation to check if a comm buffer is valid.
+
+  This function is provided so Standalone MM and Traditional MM may use a different implementation
+  of comm buffer check logic.
+
+  @param[in] Buffer  The buffer start address to be checked.
+  @param[in] Length  The buffer length to be checked.
+
+  @retval TRUE  This communicate buffer is valid per processor architecture.
+  @retval FALSE This communicate buffer is not valid per processor architecture.
+**/
+BOOLEAN
+IsCommBufferValidInternal (

```

Comment:
please refer to the https://github.com/tianocore/edk2/pull/5684/commits/3caf19de2dedf60cf5b54d9b38b1fbb86d3516d1#diff-dd35927d0476f7b6af0f598d3b37b2edf1743968ed0e4bcbd66067a16d41b656 for the buffer valid check:

XXXIsPrimaryBufferValid()         /// Cover the CommBuffer.
XXXIsNonPrimaryBufferValid() /// Cover the pointer in CommBuffer and the unblock mem

So, IsCommBufferValidInternal can be named as MmCorePerformanceIsPrimaryBufferValid() 
IsBufferOutsideMmValidInternal can be named as MmCorePerformanceIsNonPrimaryBufferValid()

Reply:
Updated the function names.